### PR TITLE
DCOS-10470: [2/4] Introducing the SearchDSLInputField

### PR DESCRIPTION
**Pull request by wavesoft**
This is the low-level `<SearchDSLInputField />` component, following closely the `<FilterInputText />` styling and props.

image

Update, adding `hasErrors` and `hasDropdown`:

image

### CODE REVIEW

**Comment by wavesoft** [comment]
**mesosphere-ci** retest this please

**Comment by Poltergeist** [comment]
LGTM 👍

File: src/js/components/SearchDSLInputField.js [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        onChange={this.handleChange}
+        ref={(ref) => this.inputField = ref}
+        type="text"
+        value={value} />
+    );
+  }
+
+  /**
+   * @override
+   */
```
Comment (rcorral) [inline src/js/components/SearchDSLInputField.js]: do we need this comment?

**Comment by rcorral** [comment]
Looks great!

File: src/js/components/SearchDSLInputField.js (hunk 1 above)
Comment (orlandohohmeier) [inline src/js/components/SearchDSLInputField.js]: I like those comments and would vote to always document overrides using JSDoc.

**Comment by orlandohohmeier** [comment]
Does #1458 needs to land first?

**Comment by jfurrow** [comment]
@wavesoft I tried to rebase your branch, but didn't realize this was pointed against a feature branch. I _believe_ I have restored this to its latest state via git's reflog, but I may have messed it up in the process. Please forgive me!!!

**Comment by wavesoft** [comment]
No worries, I don't think you did anything wrong :smile: . I introduced the `feature branch` tag to make it clear. I hope everybody is OK with this.

**Comment by wavesoft** [comment]
@jfurrow , @rcorral I created a new PR here : #1479 because for some reason the upstream branch was lost.

**Pull request closed without merging**